Customizable Discord Message and Prompts

boushrabettir opened the pull request:
### TODO
1. Create functionality for optional customizable options or both webhook message and PaLM API prompts
2. Ensure functionality works + no bugs occur during multiple testing phases
<br/>
So far the base format has been placed in the TXT files called `message.txt` and `prompt.txt`. Functionality will be placed soon.

boushrabettir committed:
Created base TXT file for customization options.

- Created base (no functionality yet) TXT files for both customized format messages, and customized PaLM API prompts.

- Updated main.py + utility.py with the changes from the fix/18 to avoid merge conflicts.

boushrabettir commented:
`Side Note`: We should not solve these merge conflicts until Fix #18 has been merged into the main branch.

boushrabettir committed:
Created db.py and customization functions in utility.py

- Created db.py file to stop the circular import (have not tested yet).

- Created the customization functions within utility.py which returns an dictionary where the values are the users custom input in the txt file.

- Updated the amount of arguments in the --days-needed cli command function to accommodate for the file paths.

- Need to implement these functions within main.py in order for the customization options to work correctly.

boushrabettir committed:
Created functionality for customization.

- Created two functions which determines the customized prompt as well as the customized message.

- Users don't need to add anything in the customization options. They can have the default output for both prompts as well as messages.

boushrabettir commented:
Users are not obligated to possess customizations for their messages and prompts. They have the freedom to selectively determine their preferences. In instances where a customized message or prompt is not furnished, the default prompt/message will be established.

**View the following results below:**

![image](https://github.com/acmcsufoss/acmOpportunities/assets/116927138/ccf99930-2ab8-4175-a576-d9ae5236f840)
![image](https://github.com/acmcsufoss/acmOpportunities/assets/116927138/c88bbfac-90f0-482a-9a62-0fb6c9d213b8)

boushrabettir commented:
**Mentioning this in order to remember before closing this PR**
When this pull request is successfully closed, we can proceed to close Issue #13.

boushrabettir committed:
Created customizable paths.

- Changed from TXT to JSON file path for customizable prompts and messages.

- Error with the Github Internship Data (will work on this on main branch).

boushrabettir committed:
Merge branch 'main' into enhancement-03

boushrabettir committed:
Updated to old cli-command and adding debugging to-dos.

Look at title.

boushrabettir committed:
Updated Github internship scraping data.

boushrabettir committed:
Removed debug comments.

boushrabettir commented:
Updated the Github web scrapping since the repository was updated (specifically the `README.md`). Customization options seem to be working well.
![image](https://github.com/acmcsufoss/acmOpportunities/assets/116927138/7cc01bfd-2304-4f44-b714-3e63c5ef19c4)

EthanThatOneKid commented on .github/workflows/automate.yaml:
```diff
@@ -37,16 +37,19 @@ jobs:
       - name: Run script
-        run: python main.py --days-needed 2 # Custom CLI Command
-#LINT.IF .env.example
+        run: python main.py --days-needed 2 PROMPTS_PATH MESSAGE_PATH
```
This command is the same command that you run locally to spin up the server.
```suggestion
        run: python main.py --days-needed 2
```

boushrabettir replied:
Awesome, thank you. I definitely forgot to update this.

EthanThatOneKid commented on db.py:
```diff
@@ -0,0 +1,29 @@
+        connection.commit()
+
+
+def delete_alL_opportunity_type(opp_type: str) -> None:
```
Where is this function used? 
```suggestion
def delete_all_opportunity_type(opp_type: str) -> None:
```

boushrabettir replied:
This is just a function for debugging, I found it useful to keep it there just in case for anything.

EthanThatOneKid replied:
> This is just a function for debugging, I found it useful to keep it there just in case for anything.

Feel free to mention this in the documentation comment of that function signature.

EthanThatOneKid commented on .github/workflows/automate.yaml:
Which one of the following is true?
- Your program uses `PROMPTS_PATH`.
- Your program uses `MESSAGE_PATH`.
- Your program uses both of these environment variables. 

What is the purpose of each environment variable used in your program?

boushrabettir replied:
The `PROMPTS_PATH` is for the customizable prompt. `MESSAGE_PATH` is for the customizable message. It was mentioned in the meeting, but I could have accidently glossed over this.

EthanThatOneKid replied:
All good; thanks for clarifying 😎

EthanThatOneKid commented on main.py:
```diff
@@ -329,5 +311,5 @@ async def main():
-if __name__ == "__main__":
-    asyncio.run(main())
+# if __name__ == "__main__":
+#     asyncio.run(main())
```
Why is this commented out? What is `asyncio.run`?

boushrabettir replied:
This is so the workflow doesn't get triggered. `asyncio.run` has been there since the beginning of this project.

EthanThatOneKid replied:
> This is so the workflow doesn't get triggered. `asyncio.run` has been there since the beginning of this project.

To temporarily disable the workflow, I’d recommend commenting out the workflow step itself.

boushrabettir replied:
For sure, will keep that in mind.

EthanThatOneKid reviewed (commented):
There’s a lot to this PR!

boushrabettir committed:
Updated README.

boushrabettir committed:
Update MAX_LIST_LENGTH and patch to PaLM.

- Created try/except error for MAX_RETRY times to ensure PaLM returns a proper list of booleans (previously the program kept throwing an attribute error).

- Decreased MAX_LIST_LENGTH to 10 since PaLM cannot handle big amounts.

boushrabettir committed:
Merge branch 'main' into enhancement-03

boushrabettir committed:
Removing comment.

EthanThatOneKid commented:
@boushrabettir, is there anything directly blocking this pull request from being merged on your end?

boushrabettir commented:
> @boushrabettir, is there anything directly blocking this pull request from being merged on your end?

Hi @EthanThatOneKid ! Yes, I have to get your approval directly.

EthanThatOneKid commented on .github/workflows/automate.yaml:
```diff
@@ -37,16 +37,19 @@ jobs:
+        run: python main.py --days-needed 2 PROMPTS_PATH MESSAGE_PATH
+        # Custom CLI Command
+
+        #LINT.IF .env.example
```
Nit: Resolve [check failure](https://github.com/acmcsufoss/acmOpportunities/actions/runs/5969957671/job/16196735903?pr=25#step:4:21).
```suggestion
#LINT.IF .env.example
```

boushrabettir replied:
Fix completed!

boushrabettir committed:
Add formatting config.

Added Black formatting config's from docs here: https://black.readthedocs.io/en/stable/the_black_code_style/current_style.html

boushrabettir committed:
Stop Workflow.

EthanThatOneKid reviewed (approved):
Looks like a lot of thought went into this PR; very nice work!

boushrabettir committed:
Remove unnecessary cli arguments.

boushrabettir committed:
Removing comment for merge.
